blob/service.go

```diff
@@ -211,83 +211,42 @@ func (s *Service) getByCommitment(
 		}
 		return nil, nil, err
 	}
+
 	for _, row := range namespacedShares {
-		if len(row.Shares) == 0 {
-			break
+		appShares, err := toAppShares(row.Shares...)
+		if err != nil {
+			return nil, nil, err
 		}
-
-		rawShares = append(rawShares, row.Shares...)
+		rawShares = append(rawShares, appShares...)
 		proofs = append(proofs, row.Proof)
 
-		// reconstruct the `blobShare` from the first rawShare in range
-		// in order to get blob's length(first share will contain this info)
-		if blobShare == nil {
-			for i, shr := range rawShares {
-				bShare, err := shares.NewShare(shr)
-				if err != nil {
-					return nil, nil, err
-				}
-
-				// ensure that the first share is not a NamespacePaddingShare
-				// these shares are used to satisfy the non-interactive default rules
-				// and are not the part of the blob, so should be removed.
-				isPadding, err := bShare.IsPadding()
-				if err != nil {
-					return nil, nil, err
-				}
-				if isPadding {
-					continue
-				}
-
-				blobShare = bShare
-				// save the length.
-				length, err := blobShare.SequenceLen()
-				if err != nil {
-					return nil, nil, err
-				}
-				amount = shares.SparseSharesNeeded(length)
-				rawShares = rawShares[i:]
-				break
-			}
-		}
-
-		// move to the next row if the blob is incomplete.
-		if amount > len(rawShares) {
-			continue
-		}
-
-		blob, same, err := constructAndVerifyBlob(rawShares[:amount], commitment)
+		var blobs []*Blob
+		blobs, rawShares, err = buildBlobsIfExist(rawShares)
```

Comment:
Could you please specify what remaining shares do you mean:
* if there will be not the last iteration of the for loop, then these remaining shares will be applied to the next rows until our blob is completed
* if you mean remaining shares on the last iteration:
Theoretically, it's not possible. All blobs must be completed, converted to the shares, and added to the data data square. So, this means that we have a bug somewhere in our code because we can't have incomplete blobs. By somewhere I mean either getter or blob service.

Reply:
If it is impossible for last iteration, let’s add check outside the loop, to make sure nothing left. So we can capture the bug if it exist. As of now if bug is there code will silently omit remaining shares.

Reply:
In the private discussion, we discussed with @walldiss two ways to handle that:
* Vlad's suggestion is to create a custom error if we have remaining shares after the loop ends. It will help users to detect the bug and properly report it.
* My opinion is that we don't need to because this is an internal bug(with storage/getters/blob-service) and returning an error will not be informative to the user. Instead, we should add a `log.Error("THIS IS A BUG. INCOMPLETE BLOB/s DETECTED") `and return add ErrBlobNotFound(if user requests blob or proof by commitment then he expects to get it back and he is almost 100% sure that this blob will be available at the particular height under the particular namespace). 

As we could not come to a consensus, we need you @Wondertan , @renaynay , @ramin

Reply:
Note that the result of `GetSharesByNamespace` gives you _verified_ and _complete_ data, meaning that this case where there are some remaining shares left is only possible if validators are malicious and made a malformed block.

Reply:
However, it would still be nice to report that to users immediately as fraud proofs may take time to propagate, and there might be bugs on our side, so some form of logging would be helpful.

On error reporting, I think we should return an error only if the user didn't get the data he requested. If there are some redundant shares, but the user got what they requested, this will be caught by FP

Reply:
This case only if the user did not get what he wanted

Reply:
If the question is to return an error or not, when the user can't get the data they want, we should definitely give him an error instead of half baked blob or blocking.